diff --git a/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/mano/record/NetworkServiceRecord.java b/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/mano/record/NetworkServiceRecord.java
index 5de7f15..38378c3 100644
--- a/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/mano/record/NetworkServiceRecord.java
+++ b/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/mano/record/NetworkServiceRecord.java
@@ -35,7 +35,7 @@ import java.util.Set;
 public class NetworkServiceRecord implements Serializable{
     @Id
     private String id;
-    @OneToMany(cascade = CascadeType.ALL)
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
     private Set<AutoScalePolicy> auto_scale_policy;
     @OneToMany(cascade = CascadeType.ALL,fetch = FetchType.EAGER)
     private Set<ConnectionPoint> connection_point;
