diff --git a/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/security/User.java b/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/security/User.java
index 83f86b9..3d869cc 100644
--- a/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/security/User.java
+++ b/openbaton-libs/catalogue/src/main/java/org/openbaton/catalogue/security/User.java
@@ -18,6 +18,7 @@ package org.openbaton.catalogue.security;
 import org.openbaton.catalogue.util.IdGenerator;
 
 import javax.persistence.*;
+import java.util.Set;
 
 @Entity
 @Table(name = "USERS")
@@ -25,17 +26,19 @@ public class User {
 
     @Id
     private String id;
-
     @Column(unique = true)
     private String username;
-
     private String password;
+    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    private Set<Role> roles;
 
-    private String firstName;
-
-    private String lastName;
+    public Set<Role> getRoles() {
+        return roles;
+    }
 
-    private boolean admin;
+    public void setRoles(Set<Role> roles) {
+        this.roles = roles;
+    }
 
     public String getId() {
         return id;
@@ -66,29 +69,13 @@ public class User {
         this.password = password;
     }
 
-    public String getFirstName() {
-        return firstName;
-    }
-
-    public void setFirstName(String firstName) {
-        this.firstName = firstName;
-    }
-
-    public String getLastName() {
-        return lastName;
+    @Override
+    public String toString() {
+        return "User{" +
+                "id='" + id + '\'' +
+                ", username='" + username + '\'' +
+                ", password='*******'" +
+                ", roles=" + roles +
+                '}';
     }
-
-    public void setLastName(String lastName) {
-        this.lastName = lastName;
-    }
-
-    public boolean isAdmin() {
-        return admin;
-    }
-
-    public void setAdmin(boolean admin) {
-        this.admin = admin;
-    }
-
-
 }
